hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/RexToExpression.java

```diff
@@ -17,39 +17,272 @@
+            case BOOLEAN:
+                return convertBooleanLiteral(literal, type);
+
+            case TINYINT:
+            case SMALLINT:
+            case INTEGER:
+            case BIGINT:
+            case DECIMAL:
+            case REAL:
+            case FLOAT:
+            case DOUBLE:
+                return convertNumericLiteral(literal, type);
+
+            case CHAR:
+            case VARCHAR:
+                return convertStringLiteral(literal, type);
+
+            case DATE:
+            case TIME:
+            case TIME_WITH_LOCAL_TIME_ZONE:
+            case TIMESTAMP:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return convertTemporalLiteral(literal, type);
+
+            case NULL:
+                return ConstantExpression.create(null, QueryDataType.NULL);
+
+            case ANY:
+                // currently, the only possible literal of ANY type is NULL
+                assert literal.getValueAs(Object.class) == null;
+                return ConstantExpression.create(null, QueryDataType.OBJECT);
+
+            default:
+                throw QueryException.error("Unsupported literal: " + literal);
+        }
+    }
+
     /**
      * Converts a {@link RexCall} to {@link Expression}.
      *
      * @param call the call to convert.
      * @return the resulting expression.
      * @throws QueryException if the given {@link RexCall} can't be
-     *                               converted.
+     *                        converted.
      */
-    @SuppressWarnings("SwitchStatementWithTooFewBranches")
+    @SuppressWarnings({"checkstyle:ReturnCount", "checkstyle:CyclomaticComplexity"})
     public static Expression<?> convertCall(RexCall call, Expression<?>[] operands) {
         SqlOperator operator = call.getOperator();
+        QueryDataType resultType = SqlToQueryType.map(call.getType().getSqlTypeName());
 
         switch (operator.getKind()) {
+            case CAST:
+                return CastExpression.create(operands[0], resultType);
+
+            case CASE:
+                return CaseExpression.create(operands, resultType);
+
+            case AND:
+                return AndPredicate.create(operands);
+
+            case OR:
+                return OrPredicate.create(operands);
+
+            case NOT:
+                return NotPredicate.create(operands[0]);
+
+            case PLUS:
+                return PlusFunction.create(operands[0], operands[1], resultType);
+
+            case MINUS:
+                return MinusFunction.create(operands[0], operands[1], resultType);
+
+            case TIMES:
+                return MultiplyFunction.create(operands[0], operands[1], resultType);
+
+            case DIVIDE:
+                return DivideFunction.create(operands[0], operands[1], resultType);
+
+            case MINUS_PREFIX:
+                return UnaryMinusFunction.create(operands[0], resultType);
+
+            case PLUS_PREFIX:
+                return operands[0];
```

Comment:
This branch is not tested.

Reply:
It's here mostly for consistency and it's a dead code: unary `+` is a no-operation in Calcite, during optimization it just strips the `+` and returns its operand. Here we do the same, just in case :)